pkg/operator/csi/csicontrollerset/csi_controller_set.go

```diff
@@ -96,7 +95,7 @@ func (c *CSIControllerSet) WithConditionalStaticResourcesController(
 	files []string,
 	shouldCreateFnArg, shouldDeleteFnArg resourceapply.ConditionalFunction,
 ) *CSIControllerSet {
-	c.conditionalStaticResourcesController = staticresourcecontroller.NewStaticResourceController(
+	c.conditionalStaticResourcesControllers = append(c.conditionalStaticResourcesControllers, staticresourcecontroller.NewStaticResourceController(
```

Comment:
Can you explain why we need to move to a slice of controllers here please?

It seems like the current expectation is that a user would only specify a single Conditional Static Resources Controller (the function doc still refers to adding a singular controller). If this expectation has changed, can you provide some information that is driving this please?

Reply:
Thank you for review , @dusk125 ! I explained it generally in commit message:

> Different assets may require different Create/Delete conditions.

And more specifically, we already have one single controller in [driver_starter.go](https://github.com/openshift/vmware-vsphere-csi-driver-operator/blob/master/pkg/operator/vspherecontroller/driver_starter.go#L82):

```
	).WithConditionalStaticResourcesController(
		"VMwareVSphereDriverConditionalStaticResourcesController",
		c.kubeClient,
		c.apiClients.DynamicClient,
		c.apiClients.KubeInformers,
		assets.ReadFile,
		[]string{
			"volumesnapshotclass.yaml",
		},
```
Create/Delete functions there are specific for VolumeSnapshotClass. Now I want to convert its StaticResourcesController:
```
	).WithStaticResourcesController(
		"VMwareVSphereDriverStaticResourcesController",
		c.kubeClient,
		c.apiClients.DynamicClient,
		c.apiClients.KubeInformers,
		assets.ReadFile,
		[]string{
			"rbac/privileged_role.yaml",
```
to ConditionalStaticResourcesController and need to specify other Create/Delete functions for them -- they will check for ManagementState==Removed.

Reply:
Thanks @mpatlasov! That makes sense and I think it's fine to allow this function to be called multiple times in order to setup multiple ConditionalControllers.

Can you update the doc comment on the WithConditional function to include that it can be called multiple times to setup multiple controllers and that it only adds and does not replace any existing CondtionalControllers please?

Reply:
Done:
```
+// The func can be called multiple times to setup multiple controllers. The calls do not replace any existing controllers.
```